port/src/maths.rs

```diff
@@ -0,0 +1,46 @@
+/// Round up by a power of 2
+pub const fn round_up2_usize(n: usize, step: usize) -> usize {
+    assert!(step.is_power_of_two());
+    (n + step - 1) & !(step - 1)
+}
+
+/// Round down by a power of 2
+pub const fn round_down2_usize(n: usize, step: usize) -> usize {
```

Comment:
It may be useful to look at something like: https://docs.rs/pow2/latest/pow2/

Reply:
I'd thought about that, but I'm a little reluctant to pull in small dependencies like this.